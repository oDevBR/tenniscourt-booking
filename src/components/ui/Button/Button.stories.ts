import { expect } from '@storybook/jest';
import { within } from '@storybook/testing-library';

import { Button } from './Button';

import type { Meta, StoryObj } from '@storybook/react';

const meta = {
  title: 'Components/Button',
  component: Button,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  args: {
    disabled: false,
  },
} satisfies Meta<typeof Button>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Basic: Story = {
  args: {
    children: 'Basic',
  },
};
Basic.play = async ({ canvasElement }) => {
  const button = within(canvasElement).getByRole('button');
  expect(button).toHaveTextContent('Basic');
  expect(button).not.toBeDisabled();
};

export const Disabled: Story = {
  args: {
    children: 'Disabled',
    disabled: true,
  },
};
Disabled.play = async ({ canvasElement }) => {
  const button = within(canvasElement).getByRole('button');
  expect(button).toHaveTextContent('Disabled');
  expect(button).toBeDisabled();
};
