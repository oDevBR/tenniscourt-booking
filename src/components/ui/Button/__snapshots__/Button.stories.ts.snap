// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Button Basic smoke-test 1`] = `
{
  "checked": undefined,
  "children": [
    {
      "checked": undefined,
      "children": undefined,
      "name": "Basic",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "components-button--basic",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
